cheroot/connections.py

```diff
@@ -55,40 +55,70 @@ class ConnectionManager:
     This is for connections which are being kept-alive for follow-up requests.
     """
 
+    _CTRL_MSG_PUT = b'P'
+    _CTRL_MSG_SHUTDOWN = b'S'
+
     def __init__(self, server):
         """Initialize ConnectionManager object.
 
         Args:
             server (cheroot.server.HTTPServer): web server object
                 that uses this ConnectionManager instance.
         """
+        self._serving = False
+        self._stop_requested = False
+        self._put_q = collections.deque()
+
         self.server = server
-        self._readable_conns = collections.deque()
         self._selector = selectors.DefaultSelector()
 
         self._selector.register(
             server.socket.fileno(),
             selectors.EVENT_READ, data=server,
         )
 
+        # create & register control socket
+        self._ctrl_rx, self._ctrl_tx = socket.socketpair()
```

Comment:
NOTE: python 3.5+ includes a pure-python implementation of this on windows, which is provided by https://pypi.org/project/backports.socketpair - let me know if that's an acceptable dependency to introduce

Reply:
Yes, using backports is reasonable. In fact, this migration to selectors has introduced such a dep recently. Just make sure to use env markers and put the try/except shim under `_compat`:
* https://github.com/cherrypy/cheroot/blame/master/setup.cfg#L68
* [ee7261b](https://github.com/cherrypy/cheroot/pull/301/commits/ee7261b6f8d1a4cf1fbbde10f311b061626d6da9) / [84f8616](https://github.com/cherrypy/cheroot/pull/301/commits/84f86162f76fd18f116822e56fd55fee38cd7bb6)